### PR TITLE
Start merging `v3` features.

### PR DESCRIPTION
**Pull request by 1uc**
This commit marks the beginning of breaking changes on `master`.

### CODE REVIEW

**Comment by 1uc** [comment]
We probably want to merge this after we've finished deprecating everything we know we want to drop. Once we merge this, we'll have a branch `v2` where we can continue to add/deprecate/backport features for `v2`.

**Review by alkino — APPROVED** [review]

**Pull request merged**
